core/MyGatewayTransportMQTTClient.cpp

```diff
@@ -258,6 +258,7 @@ bool gatewayTransportInit(void)
 	WiFi.config(_MQTT_clientIp, _gatewayIp, _subnetIp);
 #endif /* End of MY_IP_ADDRESS */
 	(void)WiFi.begin(MY_WIFI_SSID, MY_WIFI_PASSWORD, 0, MY_WIFI_BSSID);
+	delay(250);
```

Comment:
Could we document why 250 milliseconds is a good value? Does it com from a datasheet? If so, please add a comment with a reference to the datasheet and page/section. Does it come from testing? If so, could you add a comment about how the testing was performed?

If, in the future, someone thinks this number is wrong, we should be able to check why the value was chosen.

Reply:
@rikki78 I cannot reproduce the issue, see my setup & log below.
Module: ESP-WROOM-32 
Transport: RF24
ESP32 core: 1.0.3
Arduino IDE:1.8.10
MySensors: 2.3.2-beta, current dev [40d8a18](https://github.com/mysensors/MySensors/commit/40d8a186e7664896dd5f5631b7c7489d97e57954
 )
Sketch: GatewayESP32MQTTClient.ino

Log:
```
back to built-in command interpreter.
OK
>ets Jun  8 2016 00:22:57

rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0018,len:4
load:0x3fff001c,len:1100
load:0x40078000,len:9564
ho 0 tail 12 room 4
load:0x40080400,len:6320
entry 0x400806a8
37 MCO:BGN:INIT GW,CP=RNNGF---,FQ=240,REL=7,VER=2.3.2-beta
42 TSF:LRT:OK
46 TSM:INIT
49 TSF:WUR:MS=0
58 TSM:INIT:TSP OK
60 TSM:INIT:GW MODE
64 TSM:READY:ID=0,PAR=0,DIS=0
68 MCO:REG:NOT NEEDED
160 GWT:TPC:CONNECTING...
1162 MCO:BGN:STP
1163 MCO:BGN:INIT OK,TSP=1
1169 GWT:TPC:CONNECTING...
2172 TSM:READY:NWD REQ
2177 ?TSF:MSG:SEND,0-0-255-255,s=255,c=3,t=20,pt=0,l=0,sg=0,ft=0,st=OK:
2183 GWT:TPC:IP=192.168.1.165
2188 GWT:RMQ:CONNECTING...
2235 GWT:RMQ:OK
2237 GWT:TPS:TOPIC=mygateway1-out/0/255/0/0/18,MSG SENT
2696 TSF:MSG:READ,2-2-0,s=255,c=3,t=21,pt=1,l=1,sg=0:0
2701 GWT:TPS:TOPIC=mygateway1-out/2/255/3/0/21,MSG SENT

```
Please post your setup infos.

Reply:
@mfalkvidd I have only guessed this time, I can test if it can be shortened. 
@tekka007 My setup:
DOIT Esp32 DevKit v1
RF24
MySensors: 2.3.2-beta
Platform IO platform-espressif32 1.10
Sketch: GatewayESP32MQTTClient.ino

The other information I would have to check on. Please let me know what is needed or what I can test to conclude more.

Reply:
@rikki78 +full debug log, from booting up to 30s

Reply:
I couldn't replicate the issue today. I thought I was becoming crazy, but I remembered to have updated PlatformIO yesterday. And with that update, the issue is resolved.
I'm pretty sure I ran with the same configuration yesterday, but maybe a restart of Visual Code was required to work with the updated version of the ESP32 espressif software.

Just to be complete, herewith the full log. By using version 1.9.0 it still happens. So I guess we can close this issue and you can ignore the pull request.

Thanks for your support and great framework!

```
38 MCO:BGN:INIT GW,CP=RNNGF---,FQ=240,REL=7,VER=2.3.2-beta
73 TSF:LRT:OK
74 TSM:INIT
78 TSF:WUR:MS=0
87 TSM:INIT:TSP OK
89 TSM:INIT:GW MODE
93 TSM:READY:ID=0,PAR=0,DIS=0
97 MCO:REG:NOT NEEDED
371 GWT:TPC:CONNECTING...
1372 MCO:BGN:STP
1374 MCO:BGN:INIT OK,TSP=1
1532 GWT:TPC:CONNECTING...
2534 TSM:READY:NWD REQ
2539 ?TSF:MSG:SEND,0-0-255-255,s=255,c=3,t=20,pt=0,l=0,sg=0,ft=0,st=OK:
2546 TSF:MSG:READ,6-40-0,s=2,c=1,t=16,pt=0,l=1,sg=0:0
2643 GWT:TPC:CONNECTING...
3644 TSF:MSG:READ,41-41-0,s=255,c=3,t=21,pt=1,l=1,sg=0:0
3650 TSF:MSG:READ,0-41-255,s=255,c=3,t=20,pt=0,l=0,sg=0:
3657 TSF:MSG:BC
3660 TSF:MSG:READ,40-40-0,s=255,c=3,t=21,pt=1,l=1,sg=0:0
3752 GWT:TPC:CONNECTING...
4754 TSF:MSG:READ,0-30-255,s=255,c=3,t=20,pt=0,l=0,sg=0:
4759 TSF:MSG:BC
4763 TSF:MSG:READ,30-40-0,s=255,c=3,t=21,pt=1,l=1,sg=0:6
4919 GWT:TPC:CONNECTING...
6030 GWT:TPC:CONNECTING...
7032 TSF:MSG:READ,13-40-0,s=2,c=1,t=16,pt=0,l=1,sg=0:1
7208 GWT:TPC:CONNECTING...
8318 GWT:TPC:CONNECTING...
9442 GWT:TPC:CONNECTING...
10608 GWT:TPC:CONNECTING...
11717 GWT:TPC:CONNECTING...
12880 GWT:TPC:CONNECTING...
13882 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=0,sg=0:
13887 TSF:MSG:READ,4-4-0,s=5,c=2,t=3,pt=0,l=0,sg=0:
13894 TSF:MSG:READ,4-4-0,s=6,c=2,t=3,pt=0,l=1,sg=0:0
13901 TSF:MSG:ECHO REQ
13941 !TSF:MSG:SEND,0-0-4-4,s=6,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=NACK:0
13989 GWT:TPC:CONNECTING...
14992 TSF:MSG:READ,13-40-0,s=2,c=1,t=16,pt=0,l=1,sg=0:0
15219 GWT:TPC:CONNECTING...
16222 TSF:MSG:READ,13-40-0,s=2,c=1,t=16,pt=0,l=1,sg=0:1
16242 GWT:TPC:CONNECTING...
17353 GWT:TPC:CONNECTING...
18355 TSF:MSG:READ,6-40-0,s=4,c=2,t=3,pt=0,l=0,sg=0:
18360 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
18367 TSF:MSG:ECHO REQ
18373 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
18513 GWT:TPC:CONNECTING...
19515 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
19520 TSF:MSG:ECHO REQ
19526 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
19622 GWT:TPC:CONNECTING...
20624 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
20629 TSF:MSG:ECHO REQ
20635 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
20790 GWT:TPC:CONNECTING...
21792 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
21797 TSF:MSG:ECHO REQ
21803 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
21901 GWT:TPC:CONNECTING...
22903 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
22908 TSF:MSG:ECHO REQ
22914 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
23011 GWT:TPC:CONNECTING...
24013 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
24018 TSF:MSG:ECHO REQ
24024 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
24031 TSF:MSG:READ,30-40-0,s=5,c=2,t=3,pt=0,l=0,sg=0:
24038 TSF:MSG:READ,13-40-0,s=2,c=1,t=16,pt=0,l=1,sg=0:0
24245 GWT:TPC:CONNECTING...
25248 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
25253 TSF:MSG:ECHO REQ
25259 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
25280 GWT:TPC:CONNECTING...
26282 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
26287 TSF:MSG:ECHO REQ
26293 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
26430 GWT:TPC:CONNECTING...
27432 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
27437 TSF:MSG:ECHO REQ
27443 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
27540 GWT:TPC:CONNECTING...
28542 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
28547 TSF:MSG:ECHO REQ
28553 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
28560 TSF:MSG:READ,30-40-0,s=6,c=2,t=3,pt=0,l=0,sg=0:
28687 GWT:TPC:CONNECTING...
29689 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
29694 TSF:MSG:ECHO REQ
29700 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
29797 GWT:TPC:CONNECTING...
30799 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
30804 TSF:MSG:ECHO REQ
30810 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
30906 GWT:TPC:CONNECTING...
31908 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
31913 TSF:MSG:ECHO REQ
31924 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
31930 TSF:MSG:READ,6-40-0,s=2,c=1,t=16,pt=0,l=1,sg=0:0
32073 GWT:TPC:CONNECTING...
33075 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
33080 TSF:MSG:ECHO REQ
33086 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
33183 GWT:TPC:CONNECTING...
34185 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
34190 TSF:MSG:ECHO REQ
34198 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
34204 TSF:MSG:READ,30-40-0,s=4,c=2,t=3,pt=0,l=0,sg=0:
34211 TSF:MSG:READ,30-40-0,s=4,c=2,t=3,pt=0,l=2,sg=0:40
34218 TSF:MSG:ECHO REQ
34224 TSF:MSG:SEND,0-0-40-30,s=4,c=2,t=3,pt=0,l=2,sg=0,ft=0,st=OK:40
34231 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
34238 TSF:MSG:ECHO REQ
34246 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
34359 GWT:TPC:CONNECTING...
35361 TSF:MSG:READ,30-40-0,s=4,c=2,t=3,pt=0,l=2,sg=0:40
35366 TSF:MSG:ECHO REQ
35372 TSF:MSG:SEND,0-0-40-30,s=4,c=2,t=3,pt=0,l=2,sg=0,ft=0,st=OK:40
35379 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
35386 TSF:MSG:ECHO REQ
35392 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
35468 GWT:TPC:CONNECTING...
36470 TSF:MSG:READ,30-40-0,s=4,c=2,t=3,pt=0,l=2,sg=0:40
36475 TSF:MSG:ECHO REQ
36481 TSF:MSG:SEND,0-0-40-30,s=4,c=2,t=3,pt=0,l=2,sg=0,ft=0,st=OK:40
36488 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
36495 TSF:MSG:ECHO REQ
36501 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
36578 GWT:TPC:CONNECTING...
37580 TSF:MSG:READ,30-40-0,s=4,c=2,t=3,pt=0,l=2,sg=0:40
37585 TSF:MSG:ECHO REQ
37594 TSF:MSG:SEND,0-0-40-30,s=4,c=2,t=3,pt=0,l=2,sg=0,ft=0,st=OK:40
37600 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
37607 TSF:MSG:ECHO REQ
37647 !TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=NACK:0
37745 GWT:TPC:CONNECTING...
38747 TSF:MSG:READ,13-40-0,s=2,c=1,t=16,pt=0,l=1,sg=0:0
38753 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
38759 TSF:MSG:ECHO REQ
38765 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
38772 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
38778 TSF:MSG:ECHO REQ
38784 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
38855 GWT:TPC:CONNECTING...
39857 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
39862 TSF:MSG:ECHO REQ
39868 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
39875 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
39881 TSF:MSG:ECHO REQ
39887 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
40017 GWT:TPC:CONNECTING...
41019 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
41024 TSF:MSG:ECHO REQ
41030 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
41037 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
41043 TSF:MSG:ECHO REQ
41049 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
41126 GWT:TPC:CONNECTING...
42128 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
42133 TSF:MSG:ECHO REQ
42139 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
42146 TSF:MSG:READ,6-40-0,s=5,c=2,t=3,pt=0,l=1,sg=0:0
42152 TSF:MSG:ECHO REQ
42163 TSF:MSG:SEND,0-0-40-6,s=5,c=2,t=3,pt=0,l=1,sg=0,ft=0,st=OK:0
42293 GWT:TPC:CONNECTING...
```